### DISCUSSION TITLE
How to: Achieve highest zoom out

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#234] [General]
So this is not for players, just some hints of how to modify the game to make it unplayable but somewhat cool.
Also, it works when I write this. But the game sources live, things change all the time.
![synII194](https://github.com/user-attachments/assets/fea02c56-f563-4273-8951-d5e1523c861c)

Anyway:
1. Increase max render area, to make it possible to increase the range of ground tiles being rendered. Ie.:
```
diff --git a/src/enginzoom.h b/src/enginzoom.h
index 40322c7..0267def 100644
--- a/src/enginzoom.h
+++ b/src/enginzoom.h
@@ -33,5 +33,5 @@ extern "C" {
 /** Largest size of render area. Must be even to avoid shifted terrain.
  */
-#define RENDER_AREA_MAX 88
+#define RENDER_AREA_MAX 188

 #pragma pack(1)
```
2. The increase you made will overflow array of quick lights unless you intervene, ie:
```
diff --git a/src/swars.sx b/src/swars.sx
index eb02945..1657965 100644
--- a/src/swars.sx
+++ b/src/swars.sx
@@ -210773,7 +210773,7 @@ GLOBAL (word_19CC64)  /* 0x19CC64 */
 GLOBAL (word_19CC66)   /* 0x19CC66 */
                .short  0x0
 GLOBAL (super_quick_light)     /* 0x19CC68 */
-               .fill   0x4000 /* 2 * (RENDER_AREA_MAX + 1) * (RENDER_AREA_MAX + 1) */
+               .fill   0xa000 /* 2 * (RENDER_AREA_MAX + 1) * (RENDER_AREA_MAX + 1) */
 GLOBAL (tnext_screen_point)    /* 0x19EC68 */
                .short  0x0
 GLOBAL (tnext_draw_item)       /* 0x19EC6A */
```

3. Now you can render so much stuff per bucket, that bucket will overflow showing black areas in the rear. You cannot increase the amount of buckets yet, but you can increase max intems inside:
```
diff --git a/src/enginbckt.h b/src/enginbckt.h
index f934830..f9ec632 100644
--- a/src/enginbckt.h
+++ b/src/enginbckt.h
@@ -32,5 +32,5 @@ extern "C" {
 /** Max amount of draw list elements within a bucket.
  */
-#define BUCKET_ITEMS_MAX 2000
+#define BUCKET_ITEMS_MAX 6000

 #pragma pack(1)
```

4. Now you can compile the game. copy the new executable to the game. After that..

6. Edit `config/rules.ini`, Ie:
```
[engine]
ZoomMin = 44
ZoomMax = 44
```
7. Edit `config.ini`, add at end resolutions you want. Do not go above `MAX_SUPPORTED_SCREEN_WIDTH = 2560` (unless you modified that as well in the sources). Ie:
```
ResGameHi="1280x720x24"
ResGameLo="320x200x24"
ResMenu="1280x720x24"
ResFMVVidHi="640x480x24"
ResFMVidLo="320x200x24"
```

8. Run the game; if you see black cutoffs near screen edges, press `e` a few times to force-increment the render area (you will be able to increase it to the limit you've set in sources). You can also press `f` to switch to isometric mode - in isometric mode you shouldn't see black cutoffs even without any `e`s.

Congrats - your game is now unplayable, but it shows a huge part of the map at once.

**Comment by Moburma** [comment]
Nice, I will have a play with this.
